litellm/llms/sagemaker.py

```diff
@@ -160,8 +159,6 @@ def completion(
     logger_fn=None,
     acompletion: bool = False,
 ):
-    import boto3
```

Comment:
why remove this?

Reply:
These imports made the functions difficult to test- because we don't pass in the boto clients as parameters or have a module-level import, pytest won't allow mocking the module where it is imported. A possible solution to that is creating a "test" version of the entire completions or embeddings function with the mocked boto3 module and then using that as the entire function, but then I'm testing a mocked function, not the actual function. Without the mocks, I don't have the ability to tell which region is in use because there are no persistent objects to track to figure out which region was used.

Before the changes, these also made it that we would import the boto3 module upon every invocation instead of just when the proxy or library is instantiated. Boto3 won't make any network calls until a client is created and invoked, so I don't see this as a risk for impacting other operations.

My main goal for these is that we don't need to have a valid SageMaker endpoint running for unit tests to pass.

Reply:
From related conversation, I will revise 👍